composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,231 @@
+                                                conda_package='neptune',
+                                                conda_channel='conda-forge') from e
+
+        verify_type('project', project, (str, type(None)))
+        verify_type('api_token', api_token, (str, type(None)))
+        verify_type('rank_zero_only', rank_zero_only, bool)
+        verify_type('log_artifacts', log_artifacts, bool)
+        verify_type('base_namespace', base_namespace, str)
+
+        if not base_namespace:
+            raise ValueError("Argument 'base_namespace' cannot be an empty string.")
+
+        self._project = project
+        self._api_token = api_token
+        self._rank_zero_only = rank_zero_only
+        self._log_artifacts = log_artifacts
+        self._base_namespace = base_namespace
+        self._neptune_kwargs = neptune_kwargs
+
+        self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
+        self._neptune_run = None
+        self._base_handler = None
+
+        if self._enabled:
+            self._neptune_run = Run(
+                project=self._project,
+                api_token=self._api_token,
+                **self._neptune_kwargs,
+            )
+            self._base_handler = self._neptune_run[self._base_namespace]
+
+        super(NeptuneLogger, self).__init__()
+
+    @property
+    def neptune_run(self):
+        return self._neptune_run
+
+    @property
+    def base_handler(self):
+        return self._base_handler
+
+    def init(self, state: 'State', logger: 'Logger') -> None:
+        del logger  # unused
+
+        tags = [state.run_name]
+
+        if not self._rank_zero_only:
+            tags.append(f'rank{dist.get_global_rank()}')
+
+        if self._enabled:
+            assert self._neptune_run is not None
+            self._neptune_run['sys/tags'].add(tags)
+            self._neptune_run[self.INTEGRATION_VERSION_KEY] = __version__
+
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+        if not self._enabled:
+            return
+
+        assert self._base_handler is not None
+
+        self._base_handler[NeptuneLogger.METRIC_NAMESPACE].append(metrics, step=step)
+
+    def log_hyperparameters(self, hyperparameters: Dict[str, Any]) -> None:
+        if not self._enabled:
+            return
+
+        from neptune.utils import stringify_unsupported
+
+        assert self._base_handler is not None
+
+        self._base_handler[NeptuneLogger.HYPERPARAM_NAMESPACE] = stringify_unsupported(hyperparameters)
+
+    def log_traces(self, traces: Dict[str, Any]):
+        if not self._enabled:
+            return
+
+        from neptune.utils import stringify_unsupported
+
+        assert self._base_handler is not None
+
+        self._base_handler[NeptuneLogger.TRACE_NAMESPACE] = stringify_unsupported(traces)
+
+    def can_upload_files(self) -> bool:
+        """Whether the logger supports uploading files."""
+        return True
```

Comment:
This should depend on whether `upload_artifacts` is set to `True`

```suggestion
        return (self._enabled and self._upload_artifacts)
```